$OpenBSD: patch-lib-src_libnyquist_nyquist_xlisp_xlisp_h,v 1.3 2019/03/10 11:45:50 ajacoutot Exp $

- use endian.h and BYTE_ORDER
  (XXX hardcodes #define HAVE_ENDIAN_H 1)

Index: lib-src/libnyquist/nyquist/xlisp/xlisp.h
--- lib-src/libnyquist/nyquist/xlisp/xlisp.h.orig
+++ lib-src/libnyquist/nyquist/xlisp/xlisp.h
@@ -150,10 +150,17 @@ extern long ptrtoabs();
 #define XL_BIG_ENDIAN
 #endif
 
-/* Linux on Pentium */
-#if defined(__linux__) || defined(__GLIBC__)
+/* POSIX proposal
+ * Linux, BSD, etc
+ * http://austingroupbugs.net/view.php?id=162#c665
+ * Don't hardcode XL_LITTLE_ENDIAN/XL_BIG_ENDIAN
+ */
+#define HAVE_ENDIAN_H 1
+#ifdef HAVE_ENDIAN_H
+#undef XL_LITTLE_ENDIAN
+#undef XL_BIG_ENDIAN
 #include <endian.h>
-#if __BYTE_ORDER == __LITTLE_ENDIAN
+#if BYTE_ORDER == LITTLE_ENDIAN
 #define XL_LITTLE_ENDIAN
 #else
 #define XL_BIG_ENDIAN
